### PR TITLE
Add basic support for Adafruit's PyPortal board

### PR DESCRIPTION
**Pull request by Shella**
Neopixel rainbow example: ezgif-4-78dcc00b243d

### CODE REVIEW

**Comment by Shella** [comment]
Can you help us figure out the pins for the display touchscreen and WiFi/ESP32?

**Comment by sajattack** [comment]
The pins are labelled pretty well in the circuitpython board definition https://github.com/adafruit/circuitpython/blob/master/ports/atmel-samd/boards/pyportal/pins.c

File: boards/pyportal/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+use atsamd_hal as hal;
+
+#[cfg(feature = "rt")]
+pub use cortex_m_rt::entry;
+
+pub use crate::pins::Pins;
+pub use hal::target_device::*;
+pub use hal::*;
+
+use gpio::{Floating, Input, Port};
```
Comment (sajattack) [inline boards/pyportal/src/lib.rs]: Travis is complaining that Input is unused here.

File: boards/pyportal/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+use hal::clock::GenericClockController;
+use hal::sercom::{I2CMaster5, PadPin, SPIMaster2};
+use hal::time::Hertz;
+
+#[cfg(feature = "usb")]
+pub use hal::usb::UsbBus;
+#[cfg(feature = "usb")]
+use usb_device::bus::UsbBusWrapper;
+
+/// Convenience for setting up the 2x3 header block for SPI.
```
Comment (sajattack) [inline boards/pyportal/src/lib.rs]: This line of comment is referencing the 2x3 pin header on the Metro, and should be removed.

File: boards/pyportal/src/lib.rs (hunk 1 above)
Comment (sajattack) [reply boards/pyportal/src/lib.rs]: Oh, I know why, it's because it's unproven.

Comment (sajattack) [reply boards/pyportal/src/lib.rs]: lol nvm it was complaining about trellis anyway @tarcieri

**Pull request merged**